src/pastefromword.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module pastefromword/pastefromword
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import DocumentFragment from '@ckeditor/ckeditor5-engine/src/view/documentfragment';
+
+import { extractBody, bodyToView, extractStyles } from './filters/common';
+import { paragraphsToLists } from './filters/list';
+
+/**
+ * This plugin handles content pasted from Word and transforms it (if necessary)
+ * to format suitable for editor {@link module:engine/model/model~Model}.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class PasteFromWord extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'PasteFromWord';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const document = editor.editing.view.document;
+
+		this.listenTo( document, 'clipboardInput', ( evt, data ) => {
```

Comment:
Can't we make this plugin listen to `inputTransformation`? We talked that it misses some information now (`dataTransfer`), but it'd be perfect if it fitted into the existing architecture (where transformation was supposed to happen on `inputTransformation`). Otherwise, we need to think whether the architecture works fine.

Reply:
I put this to a further discussion here - https://github.com/ckeditor/ckeditor5-clipboard/issues/52#issuecomment-413853141 and then it slipped out somehow - @Reinmar  could you take a look on my linked comment, because I'm still unsure about this one.